test/UnitSession.cpp

```diff
@@ -232,6 +233,122 @@ UnitBase::TestResult UnitSession::testSlideShow()
     return TestResult::Ok;
 }
 
+/// Load a document on server and download it multiple to this client using 2 reused channels:
+/// - wsd-session for commands (websocket)
+/// - http-session to download it, as well as to load the favicon
+UnitBase::TestResult UnitSession::testSlideShowMultiDL()
+{
+    setTestname(__func__);
+    TST_LOG("Starting Test: " << testname);
+    const int dlCount = 10;
+    int dlIter = -1;
+    try
+    {
+        std::string documentPath, documentURL;
+        std::string response;
+        helpers::getDocumentPathAndURL("setclientpart.odp", documentPath, documentURL, testname);
+
+        // Reused http wsd session + socket-poll to send commands to server
+        std::shared_ptr<SocketPoll> wsdSocketPoll = std::make_shared<SocketPoll>(testname);
+        wsdSocketPoll->startThread();
+        std::shared_ptr<http::WebSocketSession> wsdSession = helpers::loadDocAndGetSession(
+            wsdSocketPoll, Poco::URI(helpers::getTestServerURI()), documentURL, testname);
+
+        // Reused http download-session for document and favicon download from server
+        std::unique_ptr<Poco::Net::HTTPClientSession> dlSession(
```

Comment:
Surely we should use Ash's new HTTP code here and elsewhere.

Also we don't need to load a document in order to do these HTTP checks. I'd suggest we don't and instead focus on the various kinds of special files we serve: capabilities, prometheus metrics etc. etc.

Reply:
I will add more test cases and expand its use case, i.e. covering the noted `FileServerRequestHandler::handleRequest(..)` and learning about the different served files. Thx!

Probably revisiting the unit test `UnitWOPIFileUrl` if suitable, as I tossed this one out after my initial attempt.
IMHO I will use the WOPI file request as used in `UnitWOPIFileUrl`, but repackage it in this linear UnitSession logic w/o the provided  WopiTestServer code.

Reply:
.. also will add the new HTTP code usage on top of the revised patch as intended earlier, to have the commits more 'on the point'.

Reply:
yeah, while working on the changeset .. I also realized its better to put the new HTTP code in and enhance the http::Header with get/set methods for `Connection`. All in progress.